compio-driver/src/poll/mod.rs

```diff
@@ -182,14 +213,26 @@ impl Driver {
         Ok(())
     }
 
-    pub fn cancel<T>(&mut self, op: Key<T>) {
+    pub fn cancel<T: crate::sys::OpCode>(&mut self, op: Key<T>) {
         self.cancelled.insert(op.user_data());
+        #[cfg(target_os = "freebsd")]
+        {
+            let mut op = op;
+            let op = op.as_op_pin();
+            if let Some(OpType::Aio(aiocbp)) = op.op_type() {
+                if let Some(aiocb) = unsafe { aiocbp.as_ptr().as_ref() } {
```

Comment:
Why not use `NonNull::as_ref`? It cannot be null afterall

Reply:
You're right. Fixed.